doc/decisions/4_in_review/change_tracking.md

```diff
@@ -174,13 +174,26 @@ The changetracking plugin needs to export at least functions for the following t
 
 ## Decision
 
+Plugins and application developer can enable changetracking via a contract.
+Store deep-duped returned keys in a separate KeySet, which we might be able to reuse as internal cache later on.
```

Comment:
Please try to also use it as cache. If it doesn't work, simply update the decision that it didn't work. To do it as separate work later is very unlikely to be ever done.

```suggestion
Store deep-duped returned keys in a separate KeySet, which we might also use as internal cache.
```

Reply:
I think this sentence needs more explanation. If "internal cache" refers to `doc/decisions/4_decided/internal_cache.md`, then the file should be linked here. Also we should mention that deep-duped means COW-copied, not fully detached copies.

Reply:
Yes, it refers to that.